src/Altinn.Platform/Altinn.Platform.Storage/Storage/Repository/TextRepository.cs

```diff
@@ -41,31 +54,70 @@ public TextRepository(IOptions<AzureCosmosSettings> cosmosettings)
                 documentCollection).GetAwaiter().GetResult();
 
             _client.OpenAsync();
+
+            _memoryCache = memoryCache;
+            _cacheEntryOptions = new MemoryCacheEntryOptions()
+                .SetPriority(CacheItemPriority.High)
+                .SetAbsoluteExpiration(new TimeSpan(0, 0, generalSettings.Value.CacheResourceLifeTimeInSeconds));
         }
 
         /// <inheritdoc/>
         public async Task<TextResource> Get(string org, string app, string language)
         {
             ValidateArguments(org, app, language);
-            try
+            string id = GetTextId(org, app, language);
```

Comment:
is the key specific enough? {org}-{app}-{language}

Reply:
I figured these would be merged and then we cache the whole thing under the app-id, but we can leave that for when it is to be implemented.